### PR TITLE
Improve telephone number validation and update error messages

### PR DESCRIPTION
**Pull request by arthurashman**
## Description of change
Updated previous regex that would allow all symbols and numeric input to allow numeric and +()- input only

Updated error messages to align with [design approved messaging ](https://docs.google.com/spreadsheets/d/1F30TsGgtB94e2AHoqo1piKqyF4_1kqbyIdKRtAFJzYQ/edit#gid=0)

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-64

### CODE REVIEW

File: config/locales/en/errors.yml [hunk 1]
```diff
@@ -29,43 +29,40 @@ en:
...
-              invalid: Please enter a valid National Insurance number
+            nino:
+              blank: Enter a National Insurance number
+              invalid: Enter a National Insurance number in the correct format
         steps/client/contact_details_form:
           attributes:
             telephone_number:
-              blank: Telephone number can’t be blank
-              invalid: Please enter a valid UK telephone number
+              blank: Enter a UK telephone number
```
Comment (zheileman) [inline config/locales/en/errors.yml:57]: Just thinking, do we want to give the impression only UK numbers are allowed? I know this is more a question for Carolina, as then the input label might need also to change.

File: config/locales/en/errors.yml [hunk 2]
```diff
@@ -29,43 +29,40 @@ en:
...
-              blank: Correspondence address can’t be blank
-              inclusion: Select a correspondence address option 
-
+              inclusion: Select a correspondence address
         steps/address/lookup_form:
           attributes:
             postcode:
               blank: Enter postcode
-              invalid: Enter a valid UK postcode
+              invalid: Enter a postcode in the correct format
```
Comment (zheileman) [inline config/locales/en/errors.yml:65]: Here, instead, we are forcibly validating it has to be a UK postcode so I think the error message can continue referencing this.

**Review by zheileman — APPROVED** [review]
👍 A few comments re: error messages but no blockers

File: config/locales/en/errors.yml (hunk 1 above)
Comment (arthurashman) [reply config/locales/en/errors.yml:57]: So I believe that is what the field always was on eForms, and throughout prototyping
@carolinapizatto - this is what we need right? We don't want to ask for any non-UK numbers?

File: config/locales/en/errors.yml (hunk 2 above)
Comment (arthurashman) [reply config/locales/en/errors.yml:65]: 'correct format' almost implies UK to my mind - as the 'correct format' will be a country specific format @carolinapizatto

Comment (zheileman) [reply config/locales/en/errors.yml:65]: Hmm, on second thought, I think the original error message `Enter a valid UK postcode` is more accurate due to:

1. in the postcode lookup (where this error happens) only UK postcodes are allowed
2. a postcode can be in a correct format in appearance and still not be a valid real postcode, i.e. `SE1 9CA`, the format looks correct to untrained eyes, but it is not really a valid postcode (user may have entered a typo in a letter).

But happy to go with whatever is decided 😄

**Pull request merged**